starboard/shared/starboard/player/filter/testing/audio_decoder_test.cc

```diff
@@ -833,6 +874,95 @@ TEST_P(AudioDecoderTest, PartialAudio) {
   }
 }
 
+TEST_P(AudioDecoderTest, ResetInEndOfStreamStateWithEndOfStream) {
+  ASSERT_NO_FATAL_FAILURE(WriteSingleInput(0));
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+
+  ResetDecoder();
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+  ASSERT_TRUE(decoded_audios_.empty());
+  ASSERT_NO_FATAL_FAILURE(AssertOutputFormatValid());
+}
+
+TEST_P(AudioDecoderTest, ResetInEndOfStreamStateWithSingleInput) {
+  ASSERT_NO_FATAL_FAILURE(WriteSingleInput(0));
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+  int decoded_audio_frames_before_reset = GetTotalFrames(decoded_audios_);
+
+  ResetDecoder();
+
+  ASSERT_NO_FATAL_FAILURE(WriteSingleInput(0));
+  WriteEndOfStream();
+  ASSERT_NO_FATAL_FAILURE(DrainOutputs());
+  ASSERT_FALSE(decoded_audios_.empty());
+  ASSERT_NO_FATAL_FAILURE(AssertOutputFormatValid());
+
+  int decoded_audio_frames_after_reset = GetTotalFrames(decoded_audios_);
```

Comment:
Instead of comparing the total frames of two outputs, do you think if we can and if it's worth to validate the contents are exactly same?

Reply:
The output contents are checked in PartialAudio tests (as it uses Reset() as well), so if the outputs are different, it will failed in PartialAudio tests.